src/realm/object-store/sync/app.cpp

```diff
@@ -1192,13 +1156,13 @@ void App::update_location_and_resend(std::unique_ptr<Request>&& request, Interme
             // Retry the original request with the updated url
             auto& request_ref = *request;
             self->m_config.transport->send_request_to_server(
-                request_ref, [self = std::move(self), completion = std::move(completion),
-                              request = std::move(request)](const Response& response) mutable {
-                    self->check_for_redirect_response(std::move(request), response, std::move(completion));
+                request_ref,
+                [completion = std::move(completion), request = std::move(request)](const Response& response) mutable {
```

Comment:
Do we need to capture self here to keep the `App` alive?

Reply:
This function is only called from `do_request()`, so the `completion` callback will have a reference to to App, like `do_request()` does.